src/utils/common/common.c

```diff
@@ -89,8 +89,7 @@ char *sstrncpy(char *dest, const char *src, size_t n) {
   return dest;
 } /* char *sstrncpy */
 
-/* ssnprintf returns zero on success, one if truncation occurred
-   and a negative integer onerror. */
+/* ssnprintf returns result from vsnprintf conistent with snprintf */
```

Comment:
so how is this function different then from snprintf?

Reply:
Hi @rubenk the ssnprintf wrapper was introduced to get around GCC 8.2/.3 warning, only, there is no functional difference.
The original suggested fix included changing return to allow handling of truncation, however as this broke stuff.
So I to allow for future handling of truncation, case I added ssnprintf2 case.
Also to ensure that the other case that had the alternated return case (in collectdctl.c), I updated this to be consistent with snprintf.
As per discussion below, this was all driven by compiler behaviour than feature/fix need and yes the amount of code touched was greater than desired.
Cheers.